test/test_module_tracker.py

```diff
@@ -76,6 +76,64 @@ def test_bw_detection(self):
             self.assertFalse(tracker.is_bw)
             self.assertEqual(tracker.parents, {"Global"})
 
+    def test_user_hooks(self):
+        class Bar(torch.nn.Module):
+            def __init__(self):
+                super(__class__, self).__init__()
+                self.foo = torch.nn.Linear(10, 10)
+
+            def forward(self, x):
+                return self.foo(x).relu_()
+
+        mt = ModuleTracker()
+        test_op = []
+
+        def hook(mod, mt, hook_name):
```

Comment:
Given how this looks like, should the first argument passed to the hook always be the module tracker?

Reply:
This is just for testing? The actual hooks have a clear signature:

1. pre_fw(mod, inp)
2. post_fw(mod, inp, op)
3. pre_bw(mod, grad_op)
4. post_bw(mod, grad_inp)

Reply:
Actually, I just modified the hook to remove any confusion.